internal/api/resources.go

```diff
@@ -203,11 +204,23 @@ func (r Resource) UpdateDBResource(res *db.Resource, info core.AssetTypeInfo) (e
 		}
 	}
 
+	isNFS := string(info.AssetType) == "nfs-shares"
 	if r.SizeConstraints == nil {
+		if isNFS {
```

Comment:
There should be no hardcoding for `nfs-shares` in this function. All asset types with a maximum size should be tested here. In particular, this line should be
```suggestion
		if maxAssetSize != nil {
```
and in the other branch the `if isNFS` can just be dropped.